skytemple_files/graphics/chara_wan/writer.py

```diff
@@ -0,0 +1,1412 @@
+    anims_node = root.find('Anims')
+    for anim_node in anims_node.iter('Anim'):
+        name = anim_node.find('Name').text
+        index = -1
+        index_node = anim_node.find('Index')
+        if index_node is not None:
+            index = int(index_node.text)
+        backref_node = anim_node.find('CopyOf')
+        if backref_node is not None:
+            backref = backref_node.text
+            if index > -1:
+                anim_stats[index] = AnimStat(index, name, None, backref)
+        else:
+            frame_width = anim_node.find('FrameWidth')
+            frame_height = anim_node.find('FrameHeight')
+            anim_stat = AnimStat(index, name, (int(frame_width.text), int(frame_height.text)), None)
+
+            rush_frame = anim_node.find('RushFrame')
+            if rush_frame is not None:
+                anim_stat.rushFrame = int(rush_frame.text)
+            hit_frame = anim_node.find('HitFrame')
+            if hit_frame is not None:
+                anim_stat.hitFrame = int(hit_frame.text)
+            return_frame = anim_node.find('ReturnFrame')
+            if return_frame is not None:
+                anim_stat.returnFrame = int(return_frame.text)
+
+            durations_node = anim_node.find('Durations')
+            for dur_node in durations_node.iter('Duration'):
+                duration = int(dur_node.text)
+                anim_stat.durations.append(duration)
+
+            if index > -1:
+                anim_stats[index] = anim_stat
+            anim_names[name.lower()] = index
+
+    copy_indices = {}
+    for idx in anim_stats:
+        stat = anim_stats[idx]
+        if stat.backref is not None:
+            back_idx = anim_names[stat.backref.lower()]
+            if back_idx not in copy_indices:
+                copy_indices[back_idx] = []
+            copy_indices[back_idx].append(idx)
+
+    # read all sheets
+    extra_sheets = []
+    anim_sheets = {}
+    for filepath in glob.glob(os.path.join(inDir, '*-Anim.png')):
+        _, file = os.path.split(filepath)
+        anim_parts = file.split('-')
+        anim_name = anim_parts[0].lower()
+        if anim_name not in anim_names:
+            extra_sheets.append(anim_name)
+        else:
+            index = anim_names[anim_name]
+            del anim_names[anim_name]
+
+            anim_img = Image.open(os.path.join(inDir, anim_name + '-Anim.png')).convert("RGBA")
+            offset_img = Image.open(os.path.join(inDir, anim_name + '-Offsets.png')).convert("RGBA")
+            shadow_img = Image.open(os.path.join(inDir, anim_name + '-Shadow.png')).convert("RGBA")
+
+            anim_sheets[index] = (anim_img, offset_img, shadow_img)
+
+    # raise warning if there exist anim stats without anims, or anims without anim stats
+    if len(anim_names) > 0:
+        orphans = []
+        for k in anim_names:
+            orphans.append(k)
+        raise Exception("Xml found with no sheet: {0}".format(', '.join(orphans)))
```

Comment:
In general please try to use more specific exceptions, such as `ValueError`, `KeyError` or `RuntimeError`, etc.